pkg/apis/management.cattle.io/v3/public_authn_types.go

```diff
@@ -229,3 +229,22 @@ type OIDCLogin struct {
 type KeyCloakOIDCProvider struct {
 	OIDCProvider `json:",inline"`
 }
+
+// +genclient
+// +kubebuilder:skipversion
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type GenericOIDCProvider struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	AuthProvider      `json:",inline"`
+
+	RedirectURL string `json:"redirectUrl"`
+	Scopes      string `json:"scopes"`
```

Comment:
For example, what format is acceptable for scopes? space-separated? comma-separated? something else?

Reply:
Our current oidc handling has been space-separated.  I suppose I could make that more clear or possibly provide support for either space or comma since I think you noted once before that comma separated is common.

Reply:
I think we can retain the space-separated, but document it.